apis/hive/v1/dnszone_types.go

```diff
@@ -27,6 +27,11 @@ type DNSZoneSpec struct {
 	// +optional
 	LinkToParentDomain bool `json:"linkToParentDomain,omitempty"`
 
+	// PreserveOnDelete allows the user to disconnect a DNSZone from Hive without deprovisioning it.
+	// This can also be used to abandon ongoing DNSZone deprovision.
+	// Typically set automatically due to PreserveOnDelete being set on a ClusterDeployment.
+	PreserveOnDelete bool `json:"preserveOnDelete,omitempty"`
```

Comment:
This is confusing and a good point, you don't have to specify this property when you create a CD, so I'm confused how that is working. Possibly omitempty does it by definition, but we should probably specify optional as well. I will do so in both places.

Reply:
It's working as long as the object is being created from go, which implicitly defaults shallow fields with their nil values. Where it would break (I think) is if a consumer used a yaml manifest with the field absent. Which is why we make a big deal about making new fields optional: so programmatic consumers can continue using old versions of manifests.

Reply:
Oh, I just read your response again. You're saying that ClusterDeployment.Spec.PreserveOnDelete is *also* not `+optional`, but it doesn't seem to be required when creating a CD -- I assume even with a yaml manifest.

So yeah, this must be due to the fact that it's a shallow value that gets its nil value when unmarshalling. Agree it's a bit odd -- does it mean there's really no way (beyond a webhook or similar) to make a shallow value required in the input?

Anyway, I guess that makes my comment moot, except that it's a signal to the consumer.

Reply:
Yeah we may as well add it, but it does not appear required in yaml for CD today. I do think the omitempty is triggering something.